### DISCUSSION TITLE
Zoom in to cursor position

### DISCUSSION THREAD

**Discussion by danieldownes** [freeplane/freeplane#1379] [Help]
When I zoom in using the mouse scroll wheel, the screen should zoom in with respect to the mouse cursor position as being the focal point of the zoom, for both in and out zooming. This makes for a better UX experience and is something well adopted in many software that users would be familar with.

**Additional context**
As a new user, I always zoom in and out when using on a smaller screen considering this software is very graphics focused. Also its one of the first things I tried to do on the help files as text can get small and scaled images can be unledgable.

I found of course you can select an element and then zoom will focus on that element. However in the case for unselectable elements, such as this image in the 'latestFreeplaneFeatures.mm', I was unable to do this, and had to select a nearby element before zooming:
![image](https://github.com/freeplane/freeplane/assets/1720388/49953046-1cee-4c74-a6d6-c80ba368f897)

By all means, retain zooming in element while using short-cut keys, however for browsing files, its more intristic to use the mouse for pan and zoom interations.

**Comment by dpolivaev** [comment]
Help->Command Search (⌘+F1)
Activate "preferences"
Type "zoom"
Discover option "Preferences…->Behaviour->Zooming keeps selected node position"
Deactivate it.
You got it.

This is a recommended way how you can find  already available Freeplane user options

**Reply by danieldownes** [reply]
Thank you. However I would say that option is misleading, as it still retains the selected node position when using the shortcut keys while deactivating.

How about having this option disabled by default to allow the origonal UX as proposed? A rename of the preference label may also be a good idea, such as 'Zoom to mouse cursor position', and invert its effectiveness.

**Reply by dpolivaev** [reply]
Considering this option when shortcut keys are used is an interesting idea. But what should Freeplane do if the mouse cursor is outside the view or just in the wrong position? I don't know. That is why this option is not considered there.

The option is a behavior change. Therefore it is not the default. Every user has their own preferences.

**Comment by danieldownes** [comment]
If you try any other visual software where pan and zoom plays a big part of interaction, then zooming to the mouse cursor is the default.
Try:
 - LibraOffice Draw
 - Krita
 - https://www.openstreetmap.org/ (or non-foss, such as Bing Maps or Google Maps).
 - GIMP
 - ...

If the mouse does not hover the active interactive area, then no scrolling happens in all these applications.

It is proposed and recommended that this is the default behaviour for scroll be inline with other visually focused apps.
